integration-testing/tests/avl-producer-api.spec.ts

```diff
@@ -0,0 +1,116 @@
+import { expect, test } from "@playwright/test";
+import { deleteDynamoItem } from "../data/dynamo";
+
+const { STAGE: stage } = process.env;
+
+const avlProducerApiUrl = `https://avl-producer.${stage}.integrated-data.dft-create-data.com`;
+const avlSubscriptionTableName = `integrated-data-avl-subscription-table-${stage}`;
+
+const testSubscription = {
+    dataProducerEndpoint: "http://siri.ticketer.org.uk/api/vm",
+    description: "Playwright test subscription",
+    shortDescription: "Playwright test subscription",
+    username: "PLAYWRIGHT",
+    password: "PLAYWRIGHT",
+    subscriptionId: "PLAYWRIGHT",
+    requestorRef: "BODS_MOCK_PRODUCER",
+    publisherId: "PLAYWRIGHT",
+};
+
+const cleardownTestSubscription = async () => {
+    await deleteDynamoItem(avlSubscriptionTableName, {
+        PK: testSubscription.subscriptionId,
+        SK: "SUBSCRIPTION",
+    });
+};
+
+test.beforeAll(async () => {
+    await cleardownTestSubscription();
+});
+
+test.afterAll(async () => {
+    await cleardownTestSubscription();
+});
+
+test.describe("avl-producer-api", () => {
+    test("should allow a new data producer subscription to be created", async ({ request }) => {
+        const newSubscription = await request.post(`${avlProducerApiUrl}/subscriptions`, {
+            data: testSubscription,
+        });
+
+        expect(newSubscription.status()).toBe(201);
+
+        const getSubscriptionResponse = await request.get(
+            `${avlProducerApiUrl}/subscriptions/${testSubscription.subscriptionId}`,
+        );
+
+        expect(getSubscriptionResponse.status()).toBe(200);
+
+        expect(await getSubscriptionResponse.json()).toEqual(
+            expect.objectContaining({
+                id: testSubscription.subscriptionId,
+                publisherId: testSubscription.publisherId,
+                status: "LIVE",
+            }),
+        );
+
+        const listSubscriptionsResponse = await request.get(`${avlProducerApiUrl}/subscriptions/`);
+
+        expect(listSubscriptionsResponse.status()).toBe(200);
+    });
```

Comment:
It would be good to include an assertion on the subscription list response like the one above checking for an object containing (and within an array in this case) the test subscription